### ISSUE TITLE
Accessibility position is not correct when using `top_level`

### ISSUE THREAD

**Issue by kitbdev** [godotengine/godot#123623]
### Tested versions

4.7.2, master
Probably since AccessKit was added.

### System information

Windows 11 (build 26200) - Multi-window, 2 monitors - OpenGL 3 (Compatibility) - NVIDIA GeForce RTX 2070 (NVIDIA; 32.0.16.1088) - Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz (12 threads) - 15.82 GiB memory - WASAPI (48000 Hz, Stereo/mono)

### Issue description

Top level is not handled correctly for accessibility.
It looks like it is offset by the parent's global transform.

<img width="515" height="260" alt="Image" src="https://github.com/user-attachments/assets/79dbfcdd-c126-4f25-82af-f628b9064517" />

### Steps to reproduce

1. Have a control that uses top level, and it has a parent that is not at the root of the scene
2. Using a screen reader, the position is offset from where it should be.

### Minimal reproduction project (MRP)

[accessibilitytoplevel_mrp.zip](https://github.com/user-attachments/files/32399367/accessibilitytoplevel_mrp.zip)
